### ISSUE TITLE
File Not Found Error

### ISSUE THREAD

**Issue by Yahoo302** [chanzer0/nfl_optimizer#10]
Not sure what I am doing wrong...any ideas?

PS C:\Users\erose\Downloads\NBA-DFS-Tools-master\NBA-DFS-Tools-master\src> python .\main.py dk opto 1 1
Traceback (most recent call last):
  File "C:\Users\erose\Downloads\NBA-DFS-Tools-master\NBA-DFS-Tools-master\src\main.py", line 70, in <module>
    main(sys.argv)
  File "C:\Users\erose\Downloads\NBA-DFS-Tools-master\NBA-DFS-Tools-master\src\main.py", line 28, in main
    opto = NBA_Optimizer(site, num_lineups, num_uniques)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\erose\Downloads\NBA-DFS-Tools-master\NBA-DFS-Tools-master\src\nba_optimizer.py", line 41, in __init__
    self.load_projections(projection_path)
  File "C:\Users\erose\Downloads\NBA-DFS-Tools-master\NBA-DFS-Tools-master\src\nba_optimizer.py", line 108, in load_projections
    with open(path) as file:
         ^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\erose\\Downloads\\NBA-DFS-Tools-master\\NBA-DFS-Tools-master\\src\\../dk_data/projections.csv'
PS C:\Users\erose\Downloads\NBA-DFS-Tools-master\NBA-DFS-Tools-master\src>

**Comment by Denwen12** [comment]
You got the projections file in the right location

**Comment by chanzer0** [comment]
```
FileNotFoundError: [Errno 2] No such file or directory: 'C:\Users\erose\Downloads\NBA-DFS-Tools-master\NBA-DFS-Tools-master\src\../dk_data/projections.csv'
```

this is telling you there is no file named `projections.csv` in the directory where it should be `\NBA-DFS-Tools\dk_data`. Add this file and it will work

**Comment by Yahoo302** [comment]
Strange because it's definitely in the right location. I wonder if it's because it's default is to open in excel.

**Comment by chanzer0** [comment]
Maybe then this is the issue: `\NBA-DFS-Tools-master\NBA-DFS-Tools-master`

you have the repository within another folder named itself. Can you post the full path of your `projections.csv` file, and what your directory structure looks like?

**Comment by Yahoo302** [comment]
projections.csv path - C:\Users\erose\Desktop\NBA-DFS-Tools-master\dk_data
(moved the dupe folder to a single folder, still same issue.
Now have the same error for the ownerships.csv but I fixed the original issue. Seems that only using Awesomo's projections work? Anyone have success using other projections?

**Comment by chanzer0** [comment]
If you're using another source for projections and ownership, just be sure they conform to the following headers

**projections.csv**
`"Name","Fpts","Position","Team","Opponent","Minutes","Salary","Pts/$","Value"`

**ownership.csv**
`"Name","Salary","Position","Matchup","Team","Opponent","Ownership %"`

**boom_bust.csv**
`"Name","Team","Position","Salary","Projection","Std Dev","Ceiling","Floor","Bust%","Boom%","Ownership%","Optimal%","Leverage"`

**Comment by chanzer0** [comment]
also be sure you're naming the file `ownership.csv` and not `ownerships.csv` (no plural)

**Comment by Yahoo302** [comment]
Thank you for helping..
Would the opto work if there is no ownership.csv & boom_bust.csv files? Projections I'm using only have regular projections.csv

**Comment by chanzer0** [comment]
The opto would still run, but you'll have incomplete data on the output. For example, if you omit `ownership.csv`, you will not have any data about the ownership of your lineup. If you omit `boom_bust.csv`, you will be missing data about a player's standard deviation range, which will make randomness obsolete. You'll also be missing data about Ceiling, Optimal, Boom and Bust scores which help dictate good from bad lineups.

In short: it should still work (you may have to comment out `self.load_boom_bust(boom_bust_path)` and `self.load_ownership(ownership_path)` in `nba_optimizer.py`) but you'll have incomplete data

**Comment by Yahoo302** [comment]
Thank you again.
Now I'm getting this error which is interesting lol

KeyError: 'Giannis Antetokounmpo'

**Comment by chanzer0** [comment]
Which line is this error occurring on? You may have a rule set in `config.json` that has Giannis, but he isn't playing today so he's not in the player pool.

**Comment by Yahoo302** [comment]
Fixed the config, getting this error now.

 File "C:\Users\erose\Desktop\NBA-DFS-Tools-master\src\nba_optimizer.py", line 359, in output
    leverage = sum(self.player_dict[player]
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\erose\Desktop\NBA-DFS-Tools-master\src\nba_optimizer.py", line 359, in <genexpr>
    leverage = sum(self.player_dict[player]
                   ^^^^^^^^^^^^^^^^^^^^^^^^
KeyError: 'Leverage'

**Comment by chanzer0** [comment]
in `load_projections` function in `nba_optimizer.py`, you will need to add a default "Leverage" key as seen in the image below.

![image](https://user-images.githubusercontent.com/25206712/209718437-f5ce2ec2-aa60-44b7-9e69-e3a206034733.png)

**Comment by Yahoo302** [comment]
adding "leverage" key gave me syntax error

Traceback (most recent call last):
  File "C:\Users\erose\Desktop\NBA-DFS-Tools-master\src\main.py", line 2, in <module>
    from nba_optimizer import *
  File "C:\Users\erose\Desktop\NBA-DFS-Tools-master\src\nba_optimizer.py", line 114
    self.player_dict[player_name] = {'Fpts': 0, 'Position': 'Leverage': 0, 'ID': 0, 'Salary': 0, 'Name': '',
                                                                      ^
SyntaxError: invalid syntax

**Comment by chanzer0** [comment]
Don't forget the comma and correct single quotes(`'` not `"`). You have some incorrect syntax somewhere

`, 'Leverage': 0,`

**Comment by Yahoo302** [comment]
Worked! Thank you so much for giving me the time and help :)

**Comment by chanzer0** [comment]
Glad I could help!
